src/pages/DetailsPage.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import { onUpdated } from 'vue'
+import { getDetails } from '@/helpers/homepageHelpers'
+import { useRoute, useRouter } from 'vue-router'
+import type { DetailedPokemon } from '../helpers/types'
+import Card from 'primevue/card'
+import Carousel from 'primevue/carousel'
+import Image from 'primevue/image'
+import Button from 'primevue/button'
+
+const route = useRoute()
+const router = useRouter()
+const pokeId = route.query.id?.toString()!
+const isshowDetails = ref(false)
+const data = ref<DetailedPokemon>({
+  name: 'bulbasaur',
+  id: 1,
+  types: [
+    { type: { name: 'grass', url: 'https://pokeapi.co/api/v2/type/12/' } },
+    { type: { name: 'poison', url: 'https://pokeapi.co/api/v2/type/4/' } }
+  ],
+  abilities: [
+    { ability: { name: 'overgrow', url: 'https://pokeapi.co/api/v2/ability/65/' } },
+    { ability: { name: 'chlorophyll', url: 'https://pokeapi.co/api/v2/ability/34/' } }
+  ],
+  sprites: {
+    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png'
+  }
+})
+
+const sprites = computed(() => {
+  const outSprites = []
+  for (const key in data.value.sprites) {
+    const sprite = data.value.sprites[key]
+    if (typeof sprite === 'string') {
+      outSprites.push(sprite)
+    }
+  }
+  console.log(outSprites)
+  return outSprites
+})
+
+const showDetails = () => {
+  isshowDetails.value = !isshowDetails.value
+}
+
+const prevButtonHandler = () => {
+  if (+pokeId === 1) {
+    router.push({ name: 'details', query: { id: 10277 } })
+  } else if (+pokeId === 10001) {
+    router.push({ name: 'details', query: { id: 1025 } })
+  } else {
+    router.push({ name: 'details', query: { id: +pokeId - 1 } })
+  }
+}
+
+const nextButtonHandler = () => {
+  if (+pokeId === 10277) {
+    router.push({ name: 'details', query: { id: 1 } })
+  } else if (+pokeId === 1025) {
+    router.push({ name: 'details', query: { id: 10001 } })
+  } else {
+    router.push({ name: 'details', query: { id: +pokeId + 1 } })
+  }
+}
+
+onUpdated(() => {
+  location.reload()
+})
+
+onMounted(async () => {
+  const d = await getDetails(pokeId)
+  console.log(d.types)
+  if (d) {
+    data.value.name = d.name
+    data.value.id = d.id
+    data.value.abilities = d.abilities
+    data.value.types = d.types
+    data.value.sprites = d.sprites
```

Comment:
Might not matter here, but always good to be careful when assigning an array to a variable, because it will assign the reference, not the contents